ApiBundle/Transformer/AreaTransformer.php

```diff
@@ -96,82 +107,101 @@ public function transform($area, NodeInterface $node = null, $parentAreaId = nul
                 $otherNode->getId()
             ));
         }
-        $facade->boDirection = $area->getBoDirection();
-
-        $facade->uiModel = $this->getTransformer('ui_model')->transform(
-            array(
-                'label' => $area->getLabel(),
-                'class' => $area->getHtmlClass(),
-                'id' => $area->getAreaId()
-            )
-        );
-
-        $facade->addLink('_self_form', $this->generateRoute('open_orchestra_backoffice_area_form', array(
-            'nodeId' => $node->getId(),
-            'areaId' => $area->getAreaId(),
-        )));
 
-        if ($this->authorizationChecker->isGranted(TreeNodesPanelStrategy::ROLE_ACCESS_UPDATE_NODE, $node)) {
-            $facade->addLink('_self_block', $this->generateRoute('open_orchestra_api_area_update_block', array(
-                'nodeId' => $node->getId(),
-                'areaId' => $area->getAreaId()
+        if ($facade->editable) {
+            $facade->addLink('_self_form_new_row', $this->generateRoute('open_orchestra_backoffice_node_new_row_area', array(
+                'nodeId' => $node->getNodeId(),
+                'language' => $node->getLanguage(),
+                'version' => $node->getVersion(),
+                'siteId' => $node->getSiteId(),
+                'areaParentId' => $area->getAreaId(),
+            )));
+            $facade->addLink('_self_move_area', $this->generateRoute('open_orchestra_api_area_move_in_node', array(
+                'areaParentId' => $area->getAreaId(),
+                'nodeId' => $node->getNodeId(),
+                'language' => $node->getLanguage(),
+                'version' => $node->getVersion(),
+                'siteId' => $node->getSiteId(),
             )));
 
-            if ($parentAreaId) {
-                $facade->addLink('_self_delete', $this->generateRoute('open_orchestra_api_area_delete_in_node_area',
-                    array(
-                        'nodeId' => $node->getId(),
-                        'parentAreaId' => $parentAreaId,
-                        'areaId' => $area->getAreaId()
-                    )
-                ));
-
-            } else {
-                $facade->addLink('_self_delete', $this->generateRoute('open_orchestra_api_area_delete_in_node',
-                    array(
-                        'nodeId' => $node->getId(),
-                        'areaId' => $area->getAreaId(),
-                    )
-                ));
+            if (AreaInterface::TYPE_COLUMN === $area->getAreaType()) {
```

Comment:
@alavieille : same remark here, a protected method could help to have a better view of the transform method